### PR TITLE
Work on various improvements connected to multi-objective optimization

### PR DESCRIPTION
**Pull request by darioizzo**
Including:
 * moving decompose_fitness out of the meta problem
 * expanding moead with the choice of decomposition weights so that it has the same capabilities of pade
 * tutorial on multiobjective optimization

### CODE REVIEW

File: include/pagmo/utils/multi_objective.hpp [hunk 1]
```diff
@@ -678,5 +678,109 @@ std::vector<vector_double> decomposition_weights(vector_double::size_type n_f, v
...
+    * was indicated as a decomposition method. Its value is ignored if "weighted" was indicated.
+    * @param method decomposition method: one of "weighted", "tchebycheff" or "bi"
+
+    *
+    * @return the decomposed objective.
+    *
+    * @throws std::invalid_argument if \p f, \p weight and \p ref_point have different sizes
+    * @throws std::invalid_argument if \p method is not one of "weighted", "tchebycheff" or "bi"
+    */
+vector_double decompose_objectives(const vector_double &f, const vector_double &weight, const vector_double &ref_point,
```
Comment (bluescarni) [inline include/pagmo/utils/multi_objective.hpp]: needs to be ``inline``.

File: include/pagmo/utils/multi_objective.hpp [hunk 2]
```diff
@@ -678,5 +678,109 @@ std::vector<vector_double> decomposition_weights(vector_double::size_type n_f, v
     return retval;
 }
 
+/// Decomposes a vector of objectives.
+/**
+    * A vector of objectives is reduced to one only objective using a decomposition
```
Comment (bluescarni) [inline include/pagmo/utils/multi_objective.hpp]: The indentation here looks funky.

File: pygmo/docstrings.cpp [hunk 3]
```diff
@@ -1625,7 +1629,8 @@ Multi Objective Evolutionary Algorithms by Decomposition (the DE variant)
 
 Raises:
     OverflowError: if *gen*, *neighbours*, *seed* or *limit* are negative or greater than an implementation-defined value
-    ValueError: if *weight_generation* is not one of 'random', 'low discrepancy', 'grid'
+    ValueError: if *decomposition* is not one of 'tchebycheff', 'weighted' or 'bi'
```
Comment (bluescarni) [inline pygmo/docstrings.cpp]: Could you replace this with a list?

Comment (darioizzo) [inline pygmo/docstrings.cpp]: Changed. But I think there are cases and this was one where the list is as easily readable as enumerating them

**Pull request merged**